pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -201,6 +205,42 @@ static void validateOptions(FhirEtlOptions options) {
 		}
 	}
 	
+	/**
+	 * Driver function for running JDBC direct fetch mode with a HAPI source server. The JDBC fetch mode
+	 * uses Beam JdbcIO to fetch FHIR resources directly from the HAPI server's database and uses the
+	 * ParquetUtil to write resources.
+	 */
+	static void runHapiJdbcFetch(FhirEtlOptions options, DatabaseConfiguration dbConfig, FhirContext fhirContext)
+	        throws PropertyVetoException {
+		Pipeline pipeline = Pipeline.create(options);
+		JdbcConnectionUtil jdbcConnectionUtil = new JdbcConnectionUtil(options.getJdbcDriverClass(),
+		        dbConfig.makeJdbsUrlFromConfig(), dbConfig.getDbUser(), dbConfig.getDbPassword(),
+		        options.getJdbcInitialPoolSize(), options.getJdbcMaxPoolSize());
+		FhirSearchUtil fhirSearchUtil = createFhirSearchUtil(options, fhirContext);
+		
+		//Get the resource count for each resource type and distribute the query workload based on batch size.
+		HashMap<String, Integer> resourceCount = (HashMap<String, Integer>) fhirSearchUtil
+		        .searchResourceCounts(options.getResourceList());
+		
+		for (String resourceType : options.getResourceList().split(",")) {
+			int numResources = resourceCount.get(resourceType);
+			
+			PCollection<List<String>> queryParameters = pipeline.apply("Generate query parameters for " + resourceType,
+			    Create.of(new JdbcFetchHapi(jdbcConnectionUtil).generateQueryParameters(options, resourceType, numResources))
+			            .withCoder(ListCoder.of(StringUtf8Coder.of())));
+			
+			PCollection<List<String>> payload = queryParameters.apply("JdbcIO fetch for " + resourceType,
+			    new JdbcFetchHapi.FetchRowsJdbcIo(
+			            JdbcIO.DataSourceConfiguration.create(jdbcConnectionUtil.getConnectionObject())));
+			
+			payload.apply("Convert to parquet for " + resourceType, ParDo.of(new ConvertResourceFn(options)));
```

Comment:
Have you looked at the `--activePeriod` feature? If you look at other `FhirEtl.run*Fetch` functions in this file you can see how it is implemented. We can leave the implementation of that for a future PR if that is too much work, but if that is the case we need to add a TODO here and also file a bug for tracking.